### ISSUE TITLE
Checking Talk Exchange Information

### ISSUE THREAD

**Issue by ajtruckle** [ajtruckle/meeting-schedule-assistant-support#51]
**Is your feature request related to a problem? Please describe.**
In Public Talks it has a [Check Talk Exchange](https://www.publictalksoftware.co.uk/help-pts/pts-check-talk-exchange.html) feature:

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/7a900659-d1bf-4737-a344-50a226d939cf)

To quote the help documentation:

> Public Talks can perform a series of **automatic checks** every time it starts. You can trigger the check **on demand** from the [Database](https://www.publictalksoftware.co.uk/help-pts/pts-db-ribbon-panel.html#CHECK) ribbon panel.
> 
> It examines the **next talk exchange** and will:
> 
> - Check to see if the home talk exchange has been **confirmed**.
> - Look for **unassigned** Home Talk numbers (i.e.:1000).

**Describe the solution you'd like**
Could a similar feature be introduced to Meeting Schedule Assistant? All the talk exchange information is held in the history database for checking. And the Congregations database Options window could be updated to include the new settings:

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/992cc539-bf1c-42ac-b496-6d6660605340)

**Additional context**
There are some new concept to be introduced in Meeting Schedule Assistant for this to work though:

1. Add reserved talk number 1000 to the talks database (1000 = unassigned).
2. Add a tentative flag to both home / away talk assignments.
3. Other reports might need updating to indicate when an exchange is tentative.

We don't have to exactly duplicate the way it is done in Public Talks but it is a basis to start from.

**Comment by ajtruckle** [comment]
@cengizu

>  I did not fully understand how the Check Talk Exchange feature works (or will work). If you can express in simpler terms what kind of ease it provides to the user (beginner level, like me), maybe then I can make suggestions.

There are two aspects to this Check …

## Tentative ##
* The Public Talks application includes a Tentative field for each home / away talk.
* What the check does is:
    *  Start application
    * Examine next home talk assignment
    * Is it still marked as tentative?
    * If yes, show the popup window:

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/7c03cccd-36d4-40ac-9f07-d81769f38324)

Think of it as a last minute way of getting you to confirm that the home talk details are correct. In Public Talks it had an extra setting to ignore home talks given by local speakers.

Does that help? I don't really know how many brothers use the Tentative field. I should also mention that any talk assignment that was tentative showed in a different colour on reports.

## Unassigned ##
In Public Talks it reserves talk number 1000 as "To Be Assigned". The idea is that you default to "1000" and then update it with the appropriate talk number for the speaker. The check in the software would check the next 1 - 4 home talks to see if the talk number was 1000 and if so flag them in a popup window. This way you can check why they are still marked as a 1000 instead of a talk number.

## Summary ##
This is how the process would work:

1. Talk Coordinator initiates a tentative exchange with a congregation for some future month.
2. All meetings are set as tentative.
3. All meetings default to 1000.
4. Closer to the time (or whenever appropriate) the Talk Coordinator confirms all the talk numbers.
5. At this point he removes the tentative flag.

So when the software starts, if the next meeting home talk is flagged as either:

* Tentative or
* 1000 Unassigned

It is an issue! Either the Talk Coordinator has not verified the exchange or has forgotten to update the schedule.

It would be nice to know if any PTS users used this feature as it would indicate if it is going to be useful for MSA.

**Comment by cengizu** [comment]
@ajtruckle 

I understand better now, thank you very much. 
This is a necessary feature, or at least it needs to be present as an option. Build a prototype and we'll test it.

**Comment by ajtruckle** [comment]
@cengizu 
I'll give it some thought for implementation.

**Comment by ajtruckle** [comment]
@cengizu 
I am in two minds about running a poll on either Telegram or the website to see how many PTS users use each of these features. 🤔

**Comment by ajtruckle** [comment]
@cengizu 
I think that some of the changes needed for this issue can also cater for issue #3 (handle symposium).

Beginning with the **Weekend Meeting Editor** window:

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/fcbb5760-59d2-4912-b11c-60f23e9bb8e3)

- Add these new items to the context menu:
    - Confirmed / Not confirmed (or To be confirmed)
    - Symposium
- Add two symbols for:
    - Indicate if confirmed or not
    - Indicate if symposium or not

**Comment by cengizu** [comment]
@ajtruckle 

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/6536357/52ae4da3-57de-4204-8895-4fc07c8670b1)

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/6536357/eefb3ca1-8293-4b39-a5ba-b35b6c2a799d)

**Comment by ajtruckle** [comment]
@cengizu 
Thanks for the mockup. 🙏

- Is the first icon meant to indicate that it is not a symposium? I was hoping to use the first icon to indicate if it was confirmed / tentative.
- Then, your second icon would be enabled / disabled if it was a symposium.
- Are you showing the word Symposium as disabled text for a reason?
- I think the menu would have a single menu item that says "Confirmed" and it would have a tick or not be ticked.
- Symposium would trigger the popup window. I'll discusses the mechanics of the popup behaviour in #3 .

**Comment by cengizu** [comment]
@ajtruckle 

Updated.

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/6536357/2362b68c-da87-4ed3-8672-16fa6e8a23d6)

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/6536357/1d179a2c-f9ce-4cff-ac66-10bd79550648)


- Confirm button, can be ON (pressed) or OFF depending user action. Same with the hamburger menu, confirmed or not confirmed state.
- The faded label shows Symposium text if there is one, otherwise no text. Just to inform user
- If there will be a symposium window, then menu entry has "..." to indicate a dialog is opening.

**Comment by ajtruckle** [comment]
@cengizu 
> Confirm button, can be ON (pressed) or OFF depending user action. Same with the hamburger menu, confirmed or not confirmed state.

When you mention a button, do you mean an interactive, clickable element rather than a mere static image?

**Comment by cengizu** [comment]
@ajtruckle 

I mention it as a button, but it can be anything that can function like on/off switch. You know better what is technically possible.

The idea is that when the user clicks, they know the speaker is confirmed, when the button/switch is not pressed, they know it's not confirmed.

**Comment by cengizu** [comment]
![Recording 2024-05-30 134757](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/6536357/670b3b15-a330-460a-a151-152d0f497bae)

A checkbox may work as well.

By default, all arrangements can be flagged as confirmed, and only tentative ones need user click.

**Comment by ajtruckle** [comment]
@cengizu 
> A checkbox may work as well.

A literal checkbox is easiest but we are then having the issue of translations. This tentative feature also needs to be on the Away Talk Entry window and we do not have the luxury of space.

What I had in mind was just a static image, either:

1. One image for confirmed
2. Another image for unconfirmed

or

1. Enabled image for confirmed
2. Disabled image for unconfirmed

The image toggles and is only for reference and not clickable. This is because the actual toggling is done via the context menu which has by-design checked menu item functionality. It will require less effort to implement.

> By default, all arrangements can be flagged as confirmed, and only tentative ones need user click.

Agreed, or the default could be added as a setting:

https://www.publictalksoftware.co.uk/help-msa/msa-congdb-options.html

**Comment by ajtruckle** [comment]
@cengizu 

Two possible icons:

* Confirmed

https://www.flaticon.com/free-icon/approval_3800787?term=agreed&page=1&position=58&origin=search&related_id=3800787
https://www.flaticon.com/free-icon-font/user-check_11271916?page=1&position=39&term=tick&origin=search&related_id=11271916

* Tentative

https://www.flaticon.com/free-icon/question_4415839?term=not+sure&page=1&position=7&origin=search&related_id=4415839

**Comment by cengizu** [comment]
@ajtruckle 

Symposium icon:
https://www.flaticon.com/free-icon/group_3394744?term=multiple&related_id=3394744

**Comment by ajtruckle** [comment]
@cengizu 
This is what I had in mind:

https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/63749c4d-3402-4d7a-8bb7-746a30037325

But the image is 24 x 24 pixels and the button is 25 x 25 pixels. This way  button margin looks nice. We can go smaller but it does not look at nice for this particular image set.

I retained the static text as a reference for comparison against the existing GUI.

**Comment by cengizu** [comment]
@ajtruckle 

Yeah, smaller size can be problematic as the icons are too dense. Simpler icons might solve this: like https://www.flaticon.com/free-icon-fonts/tick-mark

**Comment by ajtruckle** [comment]
@cengizu 
Concerning the check for **confirmed status**, I propose:

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/be1392e3-e0bb-4d80-9404-d219b2ebd66f)

* Move the Options menu from underneath Congregations to the Database menu, with a separator line.
* Add a new Start-up category to the Options window, based on the existing settings in Public Talks.
* We could also add "Number of meetings" to the set of options to let you check up to x meetings
* I just realised that this check does not include settings for Away Talk confirmed status.

**Comment by cengizu** [comment]
@ajtruckle 

Sounds too complex to me, I need to see it in action to comment.

**Comment by ajtruckle** [comment]
@cengizu 
I think that is just it ... it **sounds** too complex. But in reality I think you will find it a simple feature. Let me implement and all will become clear. 😀💪

**Comment by ajtruckle** [comment]
@cengizu 

* The **Options** menu item has been relocated from the **Congregations** sub-menu to the bottom of the **Database** menu.

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/4d2c473c-917c-4330-b6c5-68d0e1877590)

* A new **Startup** panel has been introduced at the bottom of the **Options** window.

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/028a3002-1234-4efb-b504-9209f8301c86)

See it in action:

https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/b56c5273-b6b9-43ec-801b-a36825d52a09

**Comment by ajtruckle** [comment]
@cengizu 

I have now implemented a basic check.

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/ab0338c0-e303-4128-86aa-ae92f5a02af4)

The text displayed in the popup message is also copied to the clipboard. We could opt for a custom window that lists the dates, and upon clicking them, it would display the necessary information. Alternatively, we could maintain the current simple popup. Should we decide on a custom window, I would appreciate your input.

**Comment by cengizu** [comment]
> @cengizu
> 
> * The **Options** menu item has been relocated from the **Congregations** sub-menu to the bottom of the **Database** menu.
> 
> ![image](https://private-user-images.githubusercontent.com/7985331/339801272-4d2c473c-917c-4330-b6c5-68d0e1877590.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MzQzMjQsIm5iZiI6MTcxODQzNDAyNCwicGF0aCI6Ii83OTg1MzMxLzMzOTgwMTI3Mi00ZDJjNDczYy05MTdjLTQzMzAtYjZjNS02OGQwZTE4Nzc1OTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTVUMDY0NzA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjQyZTljMzlmYTBlNWUyY2MwNWMzOWQ3ZjIxY2I3NzBjZTQ2ODc3MjhmZGY4NzdkMjI2ZDVmMTUwM2E3NTU0MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1ohDr0IfpYCUXmpZyua8EUWTfBA6mks9sjZiMRNkLc4)
> 
> * A new **Startup** panel has been introduced at the bottom of the **Options** window.
> 
> ![image](https://private-user-images.githubusercontent.com/7985331/339801689-028a3002-1234-4efb-b504-9209f8301c86.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MzQzMjQsIm5iZiI6MTcxODQzNDAyNCwicGF0aCI6Ii83OTg1MzMxLzMzOTgwMTY4OS0wMjhhMzAwMi0xMjM0LTRlZmItYjUwNC05MjA5ZjgzMDFjODYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTVUMDY0NzA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTQ5MTM0ZTRhOTQ5OGFmOTcwOGQzM2MzN2NhMDJkNzYyZjAzYjNjMGY3MDdkMzA4NjE5ZjBjNDY1OTA0YTU5NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.261JgLXIVrNbojyZW5lt4FOikwW9zDLVv8yabaj1OkU)
> 
> See it in action:
> 
>  Recording.2024-06-14.153355.mp4

This is nice

**Comment by cengizu** [comment]
> @cengizu
> 
> I have now implemented a basic check.
> 
> ![image](https://private-user-images.githubusercontent.com/7985331/339874063-ab0338c0-e303-4128-86aa-ae92f5a02af4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MzQzMjQsIm5iZiI6MTcxODQzNDAyNCwicGF0aCI6Ii83OTg1MzMxLzMzOTg3NDA2My1hYjAzMzhjMC1lMzAzLTQxMjgtODZhYS1hZTkyZjVhMDJhZjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTVUMDY0NzA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGY5M2M4ZDdiNGNmYjIwMjFjYmQ0OGQxNGNiYThlZGIwZjgzZTM4YWY5YzU4NzI1NmQ0N2Y0NjY4MWE4NzAwYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.k7kycB9tWLNLCAT8gtLe0Lg1uCGbuoxwuOlLk_5sBcA)
> 
> The text displayed in the popup message is also copied to the clipboard. We could opt for a custom window that lists the dates, and upon clicking them, it would display the necessary information. Alternatively, we could maintain the current simple popup. Should we decide on a custom window, I would appreciate your input.

We can keep the popup message for the moment, and improve it later unless you have an idea already.

**Comment by ajtruckle** [comment]
@cengizu 

> We can keep the popup message for the moment, and improve it later unless you have an idea already.

I do have some ideas but I think I will leave it like this for now. The core functionality is in place which is the main thing.

I’ll copy across existing translations from Public Talks and put in a request for the ones I need. And, I’ll also get a beta to you for testing.

**Comment by ajtruckle** [comment]
@cengizu 

Currently, MSA is set to CONFIRMED by default for backward compatibility. We can introduce a setting to specify the default state for new schedules.

## Suggestion ##
Add a 'Public Talks' section to the Database Options window, with a 'Confirmed Status' property that can be set to 'Yes' or 'No'.

For example:

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/59d11f3b-211f-4cf3-ae82-aafbefb584fb)
